archaius2-core/src/main/java/com/netflix/archaius/converters/ArrayTypeConverterFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.netflix.archaius.converters;
+
+import com.netflix.archaius.api.TypeConverter;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Type;
+import java.util.Optional;
+
+public final class ArrayTypeConverterFactory implements TypeConverter.Factory {
+    public static final ArrayTypeConverterFactory INSTANCE = new ArrayTypeConverterFactory();
+
+    private ArrayTypeConverterFactory() {}
+
+    @Override
+    public Optional<TypeConverter<?>> get(Type type, TypeConverter.Registry registry) {
+        Class clsType = (Class) type;
+
+        if (clsType.isArray()) {
+            TypeConverter elementConverter = registry.get(clsType.getComponentType()).orElseThrow(() -> new RuntimeException());
+            return Optional.of(create(elementConverter, clsType.getComponentType()));
+        }
+
+        return Optional.empty();
+    }
+
+    private static TypeConverter<?> create(TypeConverter elementConverter, Class type) {
+        return value -> {
+            String[] elements = value.split(",");
```

Comment:
any possibility that delimiter is something else? also, I'm wary of possible empty or null values. Do we protect the converter from such scenarios?

Reply:
Code here is copied from the existing version so there is no behavior change.  When converting to an array I think it's fine to fail for null or missing values.